src/map/status.cpp

```diff
@@ -13960,6 +13966,18 @@ TIMER_FUNC(status_change_timer){
 			return 0;
 		}
 		break;
+		
+	case SC_GRADUAL_GRAVITY:
+		if (--(sce->val4) >= 0) {
+			int hp = status->max_hp * sce->val2 / 100;
+
+			if (!status_charge(bl, hp, 0))
+				status_zap(bl, hp, 0);
```

Comment:
```suggestion
			status_zap(bl, hp, 0);
```
Why first try status_charge (fails if not enough HP) and then force it via status_zap (will reduce it even if it kills) in case of failure?